symengine/infinity.cpp

```diff
@@ -238,4 +238,144 @@ inline RCP<const Infty> infty(const RCP<const Number> &direction)
     return make_rcp<Infty>(direction);
 }
 
+class EvaluateInfty : public Evaluate
+{
+    virtual RCP<const Basic> sin(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(std::sin(INFINITY));
+    }
+    virtual RCP<const Basic> cos(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(std::cos(INFINITY));
+    }
+    virtual RCP<const Basic> tan(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(std::tan(INFINITY));
+    }
+    virtual RCP<const Basic> cot(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(1.0 / std::tan(INFINITY));
+    }
+    virtual RCP<const Basic> sec(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(1.0 / std::cos(INFINITY));
+    }
+    virtual RCP<const Basic> csc(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(1.0 / std::sin(INFINITY));
+    }
+    virtual RCP<const Basic> asin(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(std::asin(INFINITY));
+    }
+    virtual RCP<const Basic> acos(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(std::acos(INFINITY));
+    }
+    virtual RCP<const Basic> acsc(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(1.0 / std::asin(INFINITY));
+    }
+    virtual RCP<const Basic> asec(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(1.0 / std::acos(INFINITY));
+    }
+    virtual RCP<const Basic> atan(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<Infty>(x))
+        return number(std::atan(INFINITY));
```

Comment:
Depending on `x`'s value being `oo` and `-oo`, this should be `pi/2` and`-pi/2`. Throw a domain error otherwise.